### PR TITLE
Aggressive inlining options for `wasm-opt`

### PR DESCRIPTION
**Pull request by kentosugama**
This PR is related to the new cost model.

@luc-blaeser found in some experiments that the new overhead pertaining to the cost of `Call` instructions can be mitigated down to `~4%` with aggressive inlining options from `wasm-opt`. This has a large cost in terms of binary size (`2.2x`) but can serve as a temporary fix while we find better, Motoko specific inlining strategies.

The specific options are
```
-O4 -ifwl -aimfs 100
```

`-ifwl` stands for inline functions with loops
`-aimfs` stands for always inline max function size

The arguments to the `optimize` command has been restructured so adding further options in the future is easy.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "ic-wasm"
-version = "0.4.0"
+version = "0.5.0"
```
Comment (chenyan-dfinity) [inline Cargo.toml:3]: We are just adding new flags, no need for a major version bump?

**Review by chenyan-dfinity — APPROVED** [review]
LGTM. 

Minor point: since we have separate commands for shrink and optimize, we can also put them into separate modules.

File: Cargo.toml (hunk 1 above)
Comment (kentosugama) [reply Cargo.toml:3]: The old command no longer works so it's a breaking change.

`ic-wasm shrink --optimize` vs `ic-wasm optimize`

File: src/lib.rs [hunk 2]
```diff
@@ -2,6 +2,7 @@ pub mod info;
 pub mod instrumentation;
 pub mod limit_resource;
 pub mod metadata;
+pub mod optimize;
```
Comment (chenyan-dfinity) [inline src/lib.rs:5]:
```suggestion
#[cfg(feature = "wasm-opt")]
pub mod optimize;
```

File: src/optimize.rs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    O3,
+    #[clap(name = "O4")]
+    O4,
+    #[clap(name = "Os")]
+    Os,
+    #[clap(name = "Oz")]
+    Oz,
+}
+
+#[cfg(feature = "wasm-opt")]
```
Comment (chenyan-dfinity) [inline src/optimize.rs:24]:
```suggestion
```

**Pull request merged**